overlord/snapstate/backend/copydata.go

```diff
@@ -162,8 +162,16 @@ func (b Backend) HideSnapData(snapName string) error {
 			return fmt.Errorf("cannot create snap dir %q: %w", hiddenSnapDir, err)
 		}
 
-		// move the snap's dir
 		newSnapDir := snap.UserSnapDir(usr.HomeDir, snapName, hiddenDirOpts)
+		if exists, _, err := osutil.DirExists(newSnapDir); err != nil {
+			return err
+		} else if exists {
+			if err := os.RemoveAll(newSnapDir); err != nil {
```

Comment:
would it make sense to use the trash/untrash mechanism for this? is there an undo case to support?

Reply:
I don't think so because in that case what matters is that we move the directory back to ~/snap. The directory being overwritten shouldn't have any important data so there's really nothing to recover (AFAICT).